src/compas_timber/ghpython/components/CT_JointDef_XHalfLap/code.py

```diff
@@ -0,0 +1,40 @@
+from ghpythonlib.componentbase import executingcomponent as component
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Error
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Warning
+
+from compas_timber.connections import ConnectionSolver
+from compas_timber.connections import XHalfLapJoint
+from compas_timber.connections import JointTopology
+from compas_timber.ghpython import JointDefinition
+
+
+class XHalfLapDefinition(component):
+    def RunScript(self, MainBeam, CrossBeam):
+        max_distance_from_beams = max([MainBeam[0].width, MainBeam[0].height, CrossBeam[0].width, CrossBeam[0].height])
```

Comment:
I'd suggest moving this line to below all the input validity checks below. Otherwise it might try to calculate the distance even when there's no valid input.